### PR TITLE
Variables: Added __user.email to global variable

### LINKED ISSUE #28840: P/R #25275:  user.email was removed from global variables.
user.email was removed from global variables. 

The position taken was "there is no use case" and you wanted to limit the data exposed. 

I submit that the data is exposed in the client regardless (when logged on) and as such this only serves to limit the functionality.

The use case I have is quite simple (and clearly others do as well).   I pull data from multiple systems, many of which do not allow for federated authentication (Some use shortened logons, others require globally unique usernames across a large shared platform and don't permit email addresses as a username) 

The only useful "Key" that allows a user to be uniquely identified across the dozen or so systems is their email address. 
We are using Grafana to create user specific dashboards (ie user logs on - and they see data specific to them).   

The use of "Login Name" is not an acceptable substitute because it cannot be guaranteed to be the same in all system that we are integrating, furthermore, because we are a group of companies, the domain portion for a user may change depending on what their org unit is.   As such we cannot simply use "loginid" and append the domain portion.

Happy to re-resubmit a PR to put this back

A

### PR DESCRIPTION
**Pull request by mckn**
**What this PR does / why we need it**:
We excluded email from the global template variables to limit the amount of exposed data and most of the scenarios the `__user.id` is the same as the `__user.email`. But since explained in #28840 that might not be the case in all scenarios so I added the email as an alternative to `__user.id`. 

**Which issue(s) this PR fixes**:
Fixes #28840

**Special notes for your reviewer**:
Should this be in `7.3.2` or `7.4.0`? I guess it is a new addition to the template variable api so maybe `7.4.0`?

### CODE REVIEW

**Review by hugohaggmark — APPROVED** [review]
Looks outstanding ⭐  
(nit: e2e tests are failing though)

**Review by jackw — APPROVED** [review]
🚀

File: docs/sources/variables/variable-types/global-variables.md [hunk 1]
```diff
@@ -70,6 +70,7 @@ This variable is the ID of the current organization.
 
 `${__user.id}` is the ID of the current user.
 `${__user.login}` is the login handle of the current user.
+`${__user.email}` is the email for the current user.
```
Comment (achatterjee-grafana) [inline docs/sources/variables/variable-types/global-variables.md:73]:
```suggestion
`${__user.email}` is the email of the current user.
```

**Comment by mckn** [comment]
@hugohaggmark what do you think about the milestone? 7.4.0 or 7.3.2?

**Review by achatterjee-grafana — APPROVED** [review]
The code needs a review from dev. Added a copy-edit suggestion.

**Comment by hugohaggmark** [comment]
> @hugohaggmark what do you think about the milestone? 7.4.0 or 7.3.2?

I think 7.4.0 would do?

**Comment by oddlittlebird** [comment]
> > @hugohaggmark what do you think about the milestone? 7.4.0 or 7.3.2?
> 
> I think 7.4.0 would do?

Yes, please hold major features like these for the minor releases. It makes managing the docs much easier for @grafana/docs-squad .

**Pull request merged**